src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicPlatformDetectionTests.cs

```diff
@@ -19,5 +20,33 @@ public void UnsupportedPlatforms_ThrowsPlatformNotSupportedException()
             Assert.ThrowsAsync<PlatformNotSupportedException>(async () => await CreateQuicListener());
             Assert.ThrowsAsync<PlatformNotSupportedException>(async () => await CreateQuicConnection(new IPEndPoint(IPAddress.Loopback, 0)));
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows), nameof(PlatformDetection.SupportsTls13))]
+        public void SupportedWindowsPlatforms_IsSupportedIsTrue()
+        {
+            Assert.True(QuicListener.IsSupported);
+            Assert.True(QuicConnection.IsSupported);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsLinux))]
+        public async Task SupportedLinuxPlatforms_IsSupportedIsTrue()
+        {
+            using Process ldconfig = new Process();
+            ldconfig.StartInfo.FileName = "ldconfig";
+            ldconfig.StartInfo.Arguments = "-p";
+            ldconfig.StartInfo.RedirectStandardOutput = true;
+            ldconfig.Start();
+            string output = await ldconfig.StandardOutput.ReadToEndAsync();
+            await ldconfig.WaitForExitAsync();
+            if (output.Contains("libmsquic.so"))
+            {
+                Assert.True(QuicListener.IsSupported);
+                Assert.True(QuicConnection.IsSupported);
+            }
+            else
+            {
+                _output.WriteLine("No msquic library found.");
```

Comment:
If the above `if` condition emits two assert trues, shouldn't this `else` output an assert false?

Reply:
Nevermind. I was reading it wrong. I see what you meant.